src/pages/workspace/WorkspaceResetBankAccountModal.js

```diff
@@ -0,0 +1,75 @@
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import ConfirmModal from '../../components/ConfirmModal';
+import {cancelResetFreePlanBankAccount, resetFreePlanBankAccount} from '../../libs/actions/BankAccounts';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
+import reimbursementAccountPropTypes from '../ReimbursementAccount/reimbursementAccountPropTypes';
+import compose from '../../libs/compose';
+import ONYXKEYS from '../../ONYXKEYS';
+import bankAccountPropTypes from '../../components/bankAccountPropTypes';
+import Text from '../../components/Text';
+import styles from '../../styles/styles';
+import BankAccount from '../../libs/models/BankAccount';
+
+const propTypes = {
+    /** Reimbursement account data */
+    reimbursementAccount: reimbursementAccountPropTypes,
+
+    /** List of bank accounts */
+    bankAccountList: PropTypes.arrayOf(bankAccountPropTypes),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    reimbursementAccount: {},
+    bankAccountList: [],
+};
+
+const WorkspaceResetBankAccountModal = (props) => {
+    const isInOpenState = lodashGet(props.reimbursementAccount, 'achData.state') === BankAccount.STATE.OPEN;
+    const bankAccountID = lodashGet(props.reimbursementAccount, 'achData.bankAccountID');
+    const account = _.find(props.bankAccountList, bankAccount => bankAccount.bankAccountID === bankAccountID);
+    const bankShortName = account ? `${account.addressName} ${account.accountNumber.slice(-4)}` : '';
+    return (
+        <ConfirmModal
+            title="Are you sure?"
+            confirmText={isInOpenState ? props.translate('workspace.bankAccount.yesDisconnectMyBankAccount') : props.translate('workspace.bankAccount.yesStartOver')}
+            cancelText={props.translate('common.cancel')}
+            prompt={isInOpenState ? (
+                <Text>
+                    <Text>{props.translate('workspace.bankAccount.disconnectYour')}</Text>
+                    <Text style={styles.textStrong}>
+                        {bankShortName}
+                    </Text>
+                    <Text>{props.translate('workspace.bankAccount.bankAccountAnyTransactions')}</Text>
+                </Text>
+            ) : props.translate('workspace.bankAccount.clearProgress')}
+            danger
+            onCancel={cancelResetFreePlanBankAccount}
+            onConfirm={() => resetFreePlanBankAccount()}
```

Comment:
Should we pass the password as an argument here?

Reply:
Gonna close this one out since the password stuff is all removed.